internal/snykclient/monitordeps.go

```diff
@@ -52,7 +52,7 @@ func (t *SnykClient) MonitorDeps(
 	defer resp.Body.Close()
 
 	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
-		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+		return nil, newStatusCodeError(resp.StatusCode)
```

Comment:
what's the intention of this change? i feel the previous inline style was much more concise and readable.